operators/advanced_network/python/adv_network_common_pybind.cpp

```diff
@@ -17,109 +17,316 @@
-        "Free RX burst");
-  m.def("get_segment_packet_ptr",
-        py::overload_cast<BurstParams*, int, int>(&get_segment_packet_ptr),
-        "Get packet pointer for one segment");
-  m.def("get_packet_ptr",
-        py::overload_cast<BurstParams*, int>(&get_packet_ptr),
-        "Get packet pointer");
-  m.def("get_rx_burst", [](int port, int q) {
-      BurstParams* burst_ptr = nullptr;
-      Status status = get_rx_burst(&burst_ptr, port, q);
-      return py::make_tuple(status, py::cast(burst_ptr,
-            py::return_value_policy::take_ownership));
-      }, py::arg("port"), py::arg("q"));
-
-  // py::class_<BurstHeaderParams>(m, "BurstHeaderParams").def(py::init<>())
-  //     .def_readwrite("num_pkts",  &BurstHeaderParams::num_pkts)
-  //     .def_readwrite("port_id",   &BurstHeaderParams::port_id)
-  //     .def_readwrite("q_id",      &BurstHeaderParams::q_id);
-
-  // py::class_<BurstHeader>(m, "BurstHeader").def(py::init<>())
-  //     .def_readwrite("hdr",  &BurstHeader::hdr);
-
-  // py::class_<BurstParams>(m, "BurstParams").def(py::init<>())
-  //     .def_readwrite("hdr", &BurstParams::hdr)
-  //     .def_readwrite("cpu_pkts", &BurstParams::cpu_pkts)
-  //     .def_readwrite("gpu_pkts", &BurstParams::gpu_pkts);
-
-  py::class_<BurstHeaderParams>(m, "BurstHeaderParams").def(py::init<>());
-  py::class_<BurstHeader>(m, "BurstHeader").def(py::init<>());
-  py::class_<BurstParams>(m, "BurstParams").def(py::init<>());
-  //  py::class_<BurstParams, std::shared_ptr<BurstParams>>
-  //    (m, "BurstParams").def(py::init<>());
+        "Set parameters of burst header",
+        "burst"_a,
+        "port"_a,
+        "q"_a,
+        "num"_a,
+        "segs"_a);
+
+  // Memory management functions
+  m.def("free_all_segment_packets",
+        py::overload_cast<BurstParams*, int>(&free_all_segment_packets),
+        "Free all packets in a burst for one segment",
+        "burst"_a,
+        "seg"_a);
+  m.def("free_all_packets_and_burst_rx",
+        py::overload_cast<BurstParams*>(&free_all_packets_and_burst_rx),
+        "Free all packets and burst structure for RX",
+        "burst"_a);
+  m.def("free_all_packets_and_burst_tx",
+        py::overload_cast<BurstParams*>(&free_all_packets_and_burst_tx),
+        "Free all packets and burst structure for TX",
+        "burst"_a);
+  m.def("free_segment_packets_and_burst",
+        py::overload_cast<BurstParams*, int>(&free_segment_packets_and_burst),
+        "Free all packets and burst structure for one packet segment",
+        "burst"_a,
+        "seg"_a);
+  m.def(
+      "free_tx_burst", py::overload_cast<BurstParams*>(&free_tx_burst), "Free TX burst", "burst"_a);
+  m.def(
+      "free_rx_burst", py::overload_cast<BurstParams*>(&free_rx_burst), "Free RX burst", "burst"_a);
+
+  // Network interface functions
+  m.def("get_mac_addr",
+        py::overload_cast<int, char*>(&get_mac_addr),
+        "Get MAC address of an interface",
+        "port"_a,
+        "mac"_a);
```

Comment:
Writes to the provided char* will not work from Python. Consider returning a tuple where the second element is the result formatted as a Python string (empty string if the status was not successful).

```suggestion
// Return formatted string (most common use case)
m.def("get_mac_addr",
      [](int port) -> py::tuple {
        char mac[6];
        Status status = get_mac_addr(port, mac);
        if (status == Status::SUCCESS) {
          char formatted[18];
          snprintf(formatted, sizeof(formatted), 
                   "%02x:%02x:%02x:%02x:%02x:%02x",
                   static_cast<unsigned char>(mac[0]),
                   static_cast<unsigned char>(mac[1]),
                   static_cast<unsigned char>(mac[2]),
                   static_cast<unsigned char>(mac[3]),
                   static_cast<unsigned char>(mac[4]),
                   static_cast<unsigned char>(mac[5]));
          return py::make_tuple(status, std::string(formatted));
        }
        return py::make_tuple(status, std::string(""));
      },
      "Get MAC address of an interface as formatted string",
      "port"_a);
```

Reply:
Updated, according to the suggestion

Reply:
Changes made as suggested."